client/purplecaffeine/backend.py

```diff
@@ -20,14 +22,40 @@ def __init__(self, path: str):
         """Local backend.
 
         Args:
-            path: folder where trail data will be saved.
+            path: folder where trial data will be saved.
         """
         self.path = path
 
-    def save_trial(self, trial):
+    def save_trial(self, name: str, trial_json: json) -> str:
         """Saves given trial.
 
         Args:
-            trial: trial to save
+            name: name of the trial
+            trial_json: encode trial to save
+
+        Returns:
+            self.path: path of the trial file
         """
+
+        with open(
+            self.path + "/" + name + ".json", "w", encoding="utf-8"
+        ) as trial_file:
+            trial_file.write(trial_json)
```

Comment:
Maybe we should use `json.dump` with `JsonDecoder` for Trial. What do you think? Something like

```python
class OurCustomEncoderForTrial(JsonEncoder):
     """Encoder which serialize content of Trial into json/."""

with open() as f:
    json.dump(f, trial, cls=OurCustomEncoderForTrial)
```

Reply:
It was what I was thinking at first but as the style object of Qiskit keep changing overtime I wouldn't want for us to have to often deprecated format. Furthermore, we indeed style have tuple object...
Let's try something, you'll tell me :)